src/routing/mod.rs

```diff
@@ -206,15 +248,53 @@ pub async fn setup_router(
             .map(ConcurrencyLimitLayer::new),
     );
 
-    // Load shedder
-    let load_shedder_mw = option_layer(cli.load.load_shed_ewma_param.map(|x| {
-        ServiceBuilder::new()
-            .layer(MapResponseLayer::new(|resp| match resp {
-                LoadShedResponse::Inner(inner) => inner,
-                LoadShedResponse::Overload => ErrorCause::LoadShed.into_response(),
-            }))
-            .layer(LoadShedLayer::new(x, cli.load.load_shed_target_latency))
-    }));
+    // Load shedders
+
+    // We need to map the generic response of a shedder to an Axum's Response
+    let shed_map_response = MapResponseLayer::new(|resp| match resp {
+        ShedResponse::Inner(inner) => inner,
+        ShedResponse::Overload(_) => ErrorCause::LoadShed.into_response(),
+    });
+
+    let load_shedder_system_mw = option_layer(
+        // TODO make nicer?
```

Comment:
I'm assuming the types are the same, in which case you can do something like this:
```rs
use anyhow::Error;

#[tokio::main]
async fn main() -> Result<(), Error> {
    let vs: Vec<Option<i32>> = vec![Some(1), None, None, None];

    if vs.iter().any(|&x| x.is_some()) {
        println!("1");
    }

    Ok(())
}
```

Reply:
Yeah, but putting all that in a vector probably would be even more verbose... 😕